app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt

```diff
@@ -134,28 +147,45 @@ object ThreadController {
         }
     }
 
-    // TODO: Replace this with a RealmList sub query (blocked by https://github.com/realm/realm-kotlin/issues/1037)
+    // TODO: Replace this with a RealmList sub query.
     fun getThreadUnseenMessagesUids(thread: Thread): List<String> {
-        return mutableListOf<String>().apply {
-            thread.messages.forEach { message ->
-                if (!message.seen) {
-                    add(message.uid)
-                    addAll(thread.getMessageDuplicatesUids(message.messageId))
-                }
-            }
-        }
+        return getThreadMessagesAndDuplicatesUids(thread) { message -> !message.seen }
+    }
+
+    // TODO: Replace this with a RealmList sub query.
+    fun getThreadUnseenMessages(thread: Thread): List<Message> {
+        return getThreadMessagesAndDuplicates(thread) { message -> !message.seen }
```

Comment:
So the only difference between these two methods is the type of return that changes.
But except that it makes a dupplication of logic, and also especially when we will pass to the `RealmList subquery` the other method will be useless because this one will just return objects.

And also we make also dupplications of logic and also especially the met🤔hod which returns the uids is not even used because it is already treated with a map in `MainViewModel`

Reply:
I removed the `...Uids()` versions.

This refactor is in another PR: https://github.com/Infomaniak/android-mail/pull/470